cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -484,7 +475,118 @@ private void deletePVCs() {
     return resultFuture;
   }
 
+  /**
+   * Deletes the job controlled by this controller asynchronously.
+   *
+   */
+  private void deleteJob() {
```

Comment:
Why is it not returning a future? We should be following the same pattern as for deployment and statefulset.